packages/@atjson/document/test/serialize-test.ts

```diff
@@ -0,0 +1,1129 @@
+
+describe("serialize", () => {
+  describe("blocks", () => {
+    test("single block", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "Hello, world",
+            annotations: [
+              new Paragraph({
+                start: 0,
+                end: 12,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCHello, world",
+        blocks: [
+          {
+            type: "paragraph",
+            attributes: {},
+          },
+        ],
+        marks: [],
+      });
+    });
+
+    test("colinear marks & blocks", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "Hello, world",
+            annotations: [
+              new Paragraph({
+                start: 0,
+                end: 12,
+              }),
+              new Italic({
+                start: 0,
+                end: 12,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCHello, world",
+        blocks: [
+          {
+            type: "paragraph",
+            attributes: {},
+          },
+        ],
+        marks: [
+          {
+            type: "italic",
+          },
+        ],
+      });
+    });
+
+    test("colinear marks", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "Hello, world",
+            annotations: [
+              new Paragraph({
+                start: 0,
+                end: 12,
+              }),
+              new Italic({
+                start: 0,
+                end: 12,
+              }),
+              new Bold({
+                start: 0,
+                end: 12,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCHello, world",
+        blocks: [
+          {
+            type: "paragraph",
+            attributes: {},
+          },
+        ],
+        marks: [
+          {
+            type: "bold",
+            range: "(1..13]",
+          },
+          {
+            type: "italic",
+            range: "(1..13]",
+          },
+        ],
+      });
+    });
+
+    test("objects", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "Missy Elliott’s\uFFFC“Supa Dupa Fly”",
+            annotations: [
+              new LineBreak({
+                start: 15,
+                end: 16,
+              }),
+              new ParseAnnotation({
+                start: 15,
+                end: 16,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCMissy Elliott’s\uFFFC“Supa Dupa Fly”",
+        blocks: [
+          {
+            type: "text",
+            parents: [],
+          },
+          {
+            type: "line-break",
+            selfClosing: true,
+            parents: ["text"],
+            attributes: {},
+          },
```

Comment:
I see we're treating line breaks (all object annotations?) as children of `text` rather than as siblings. I know there was some discussion about this - what was the rationale for going with this approach? Do other formats do it this way? The ones I can immediately think of (DOM, JSONML) would have these as siblings

Reply:
If we treated a line break as a block (that is `selfClosing: false`), then this would be the behavior that you're describing. We can definitely do that, but I have thought of a line break as _in_ a block, and I believe that text editors like prosemirror think of it this way as well.

`text` here is more of a structural implementation detail such that there's always a parent block in a document. It kind of is analogous to text nodes in the DOM, but we're only using it for disambiguation purposes rather than _everywhere_. This is a specific case that needs disambiguation— we need at least _one_ block in every document per the design of the serialization format. (Theoretically, we could have zero, but it serves us to always have one for implementation with things like prosemirror and other editors that require blocks)